docs/articles/documentation/extending-testcafe/custom-browser-provider-plugin/index.md

```diff
@@ -0,0 +1,200 @@
+                browserInfo.cmd  = '--new-window';
+                break;
+            default:
+                throw new Error('Unsupported browser!');
+        }
+
+        await browserTools.open(browserInfo, pageUrl);
+    },
+
+    async closeBrowser (id) {
+        await browserTools.close(id);
+    },
+
+    async getBrowserList () {
+        return ['firefox', 'chrome'];
+    },
+
+    async isValidBrowserName (browserName) {
+        var browserList = await this.getBrowserList();
+
+        return browserList.indexOf(browserName) > -1;
+    },
+
+    async resizeWindow (id, width, height, currentWidth, currentHeight) {
+        await browserTools.resize(id, currentWidth, currentHeight, width, height);
+    },
+
+    async takeScreenshot (id, screenshotPath) {
+        await browserTools.screenshot(id, screenshotPath);
+    },
+};
+```
+
+> To learn how to implement provider methods, you can also explore the existing browser provider plugins, for example, [testcafe-browser-provider-saucelabs](https://github.com/DevExpress/testcafe-browser-provider-saucelabs/) or [testcafe-browser-provider-phantomjs](https://github.com/DevExpress/testcafe-browser-provider-phantomjs).
+
+## Building the Provider
+
+You can build a browser provider project by using the `build` Gulp task.
+
+```bash
+gulp build
+```
+
+## Using the Provider Development Version
+
+If you are still developing the browser provider, but need to test it within TestCafe, there is no need to publish the browser provider package to npm.
+You can link the browser provider to TestCafe by using the [npm link](https://docs.npmjs.com/cli/link) command.
+This allows you to work on the browser provider project and test it iteratively without having to re-publish everytime you make a change.
+
+To link the browser provider package, navigate to the provider directory and run `npm link`:
+
+```bash
+cd my-provider
+npm link
+```
+
+After that TestCafe will use the provider version you are currently developing.
+
+For information on how to specify a browser in tests, see [Using the Browser Provider](../../using-testcafe/common-concepts/using-browsers.md#using-the-browser-providers).
+
+## Publishing the Provider to NPM
+
+When you finish developing the provider, you can publish it to npm by running the `publish-please` npm script.
+This script builds the package, tests the provider and then uses [publish-please](https://github.com/inikulin/publish-please) to publish it to npm.
+That is why it is recommended to use the `publish-please` script instead of `npm publish`.
+
+```bash
+npm run publish-please
+```
+
+After that you can install the provider plugin as you would [install any other plugin](../index.md#installing-plugins) and use it in the same manner as you would use another [browser providers](../../using-testcafe/common-concepts/using-browsers.md#using-the-browser-providers).
+
+## Reference
```

Comment:
Do we really need this paragraph since we give a link to `Browser Provider Methods` in the `Implementing the Browser Provider` above?